openshift-kube-apiserver/admission/customresourcevalidation/route/validate_route.go

```diff
@@ -45,18 +41,21 @@ func toRoute(uncastObj runtime.Object) (*routev1.Route, field.ErrorList) {
 }
 
 type routeV1 struct {
+	sarc           authorizationv1client.SubjectAccessReviewInterface
+	secretsGetter  corev1client.SecretsGetter
+	validationOpts routecommon.RouteValidationOptions
 }
 
-func (routeV1) ValidateCreate(obj runtime.Object) field.ErrorList {
+func (r routeV1) ValidateCreate(obj runtime.Object) field.ErrorList {
 	routeObj, errs := toRoute(obj)
 	if len(errs) > 0 {
 		return errs
 	}
 
-	return routevalidation.ValidateRoute(routeObj)
+	return routevalidation.ValidateRoute(context.TODO(), routeObj, r.sarc, r.secretsGetter, r.validationOpts)
```

Comment:
I think that we should extend `ObjectValidator` to accept the `ctx` for all its methods.

```
type ObjectValidator interface {
	ValidateCreate(obj runtime.Object) field.ErrorList
	ValidateUpdate(obj runtime.Object, oldObj runtime.Object) field.ErrorList
	ValidateStatusUpdate(obj runtime.Object, oldObj runtime.Object) field.ErrorList
}
```

Reply:
Addressed